core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -315,7 +314,7 @@ private[spark] class ExecutorAllocationManager(
   private def onExecutorAdded(executorId: String): Unit = synchronized {
     if (!executorIds.contains(executorId)) {
       executorIds.add(executorId)
-      executorIds.foreach(onExecutorIdle)
+      onExecutorIdle(executorId)
```

Comment:
> But its expireTime actually is already removed here, so if we don't mark it idle when a new executor joins, it will never be removed.

But I have changed these codes to:

```

    removeTimes.retain { case (executorId, expireTime) =>
      now < expireTime || !removeExecutor(executorId)
    }
```
It won't remove `(executorId, expireTime)` if `removeExecutor(executorId)` return false.